src/compiler/checker.ts

```diff
@@ -1020,38 +1020,68 @@ namespace ts {
             return links.resolvedExports || (links.resolvedExports = getExportsForModule(moduleSymbol));
         }
 
-        function extendExportSymbols(target: SymbolTable, source: SymbolTable) {
+        interface ExportStarDiagnosticsLookupTable {

```

Comment:
Also, the text isn't the id of the symbol in the map - the map key id is the duplicated symbol's id, the text is the text of the export specifier it was first found in.